packages/@react-aria/spinbutton/src/useSpinButton.ts

```diff
@@ -168,13 +166,12 @@ export function useSpinButton(
         initialStepDelay
       );
     },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [onDecrement, minValue, value]
+    [onDecrement, minValue, value, clearAsync, _async]
```

Comment:
I seem to remember having issues with event handlers getting out of date, especially with the timers firing them. For example, the `window.setTimeout` here recursively calls `onDecrementPressStart`, and that will not be the new instance of it - it'll be the one at the time it was originally called. So then it would be calling the wrong version of `onDecrement`.

Reply:
Yeah, I had to take some extra steps to account for that
https://github.com/adobe/react-spectrum/pull/4005/files#diff-f920ff47503335092fb8b412b9e4d0679256fecedc505827ad70a7c7871354dbR164
see spinRef

the setState function is stable, so no matter what copy of onDecrement is called, it will always call the same setState

and then the portion that determines the next value uses the spin ref, so again, doesn't matter what copy of the function I call, and I think this is safe for reasons I outlined here https://github.com/adobe/react-spectrum/pull/4005#discussion_r1129035366

happy to use something else if you can think of another solution